### PR TITLE
[PW-7716] Removing unused depedencies

### PR DESCRIPTION
**Pull request by jillingk**
**Description**
Removed unused dependencies and removed some Jackson specific annotations from MarketPay endpoints.

**Tested scenarios**
UT's

### CODE REVIEW

**Comment by jillingk** [comment]
> @jillingk Do we need this [dependency](https://github.com/Adyen/adyen-java-api-library/blob/3f59a2868f2fbd3bc4c61a21598bdbfbc59695c3/pom.xml#L236) ?

So yeah no I don't think we really do. It's this nullable annotation we do in some generated parts of the library, but it's only really for IDE's to recognise nullpointerexceptions or smth? Either way I removed it (through the templates) and everything still worked fine. But I did want to discuss first before I removed it in this PR.
```java
@javax.annotation.Nonnull
```

**Comment by jillingk** [comment]
Okio is for the ByteString type in the ByteArrayAdapter in the *.JSON classes:
```java
@Override
        public byte[] read(JsonReader in) throws IOException {
            switch (in.peek()) {
                case NULL:
                    in.nextNull();
                    return null;
                default:
                    String bytesAsBase64 = in.nextString();
                    ByteString byteString = ByteString.decodeBase64(bytesAsBase64);
                    return byteString.toByteArray();
            }
        }
```
It's generated w the templates, but if we can find an (java native) alternative I'm sure we can change it.

File: src/main/java/com/adyen/model/marketpay/AccountEvent.java [hunk 1]
```diff
@@ -52,7 +51,7 @@ public enum EventEnum {
         @SerializedName("RefundNotPaidOutTransfers")
         REFUNDNOTPAIDOUTTRANSFERS("RefundNotPaidOutTransfers");
 
-        @JsonValue
+
```
Comment (wboereboom) [inline src/main/java/com/adyen/model/marketpay/AccountEvent.java:54]: can we not have this blank line?

**Review by wboereboom — COMMENTED** [review]
Have we made sure removing these did not break any enum (de)serialization?

**Review by AlexandrosMor — DISMISSED** [review]

**Comment by michaelpaul** [comment]
> Okio is for the ByteString type in the ByteArrayAdapter in the *.JSON classes:
> 
> ```java
> @Override
>         public byte[] read(JsonReader in) throws IOException {
>             switch (in.peek()) {
>                 case NULL:
>                     in.nextNull();
>                     return null;
>                 default:
>                     String bytesAsBase64 = in.nextString();
>                     ByteString byteString = ByteString.decodeBase64(bytesAsBase64);
>                     return byteString.toByteArray();
>             }
>         }
> ```
> 
> It's generated w the templates, but if we can find an (java native) alternative I'm sure we can change it.

Let's take a look into https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html.

**Comment by jillingk** [comment]
> > Okio is for the ByteString type in the ByteArrayAdapter in the *.JSON classes:
> > ```java
> > @Override
> >         public byte[] read(JsonReader in) throws IOException {
> >             switch (in.peek()) {
> >                 case NULL:
> >                     in.nextNull();
> >                     return null;
> >                 default:
> >                     String bytesAsBase64 = in.nextString();
> >                     ByteString byteString = ByteString.decodeBase64(bytesAsBase64);
> >                     return byteString.toByteArray();
> >             }
> >         }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It's generated w the templates, but if we can find an (java native) alternative I'm sure we can change it.
> 
> Let's take a look into https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html.

I actually ended up using the apache library, that we already used, to do this encoding (only one line of code compared to some other solutions). Also moved the okio to test scope, and generated the JSON.java files so it now does not use okio anymore.

File: pom.xml [hunk 2]
```diff
@@ -210,31 +200,20 @@
...
         <dependency>
             <groupId>io.swagger.core.v3</groupId>
             <artifactId>swagger-annotations</artifactId>
             <version>2.2.8</version>
         </dependency>
         <dependency>
-            <groupId>javax.ws.rs</groupId>
-            <artifactId>javax.ws.rs-api</artifactId>
-            <version>2.1.1</version>
+            <groupId>jakarta.ws.rs</groupId>
```
Comment (wboereboom) [inline pom.xml:214]: lets look into this first before just changing it. Are we sure we need to do this?

**Review by wboereboom — COMMENTED** [review]
do we test base64 (de)serialisation in unit tests?

**Comment by jillingk** [comment]
> do we test base64 (de)serialisation in unit tests?

Yep https://github.com/Adyen/adyen-java-api-library/blob/bd9cd6338aa9fdffb745b5dc20e442503465e54e/src/test/java/com/adyen/PaymentTest.java#L460

File: pom.xml (hunk 2 above)
Comment (jillingk) [reply pom.xml:214]: It's actually just the same package but under a [new name](https://mvnrepository.com/artifact/javax.ws.rs/javax.ws.rs-api).

File: pom.xml [hunk 3]
```diff
@@ -210,31 +200,20 @@
             <version>4.11.0</version>
             <scope>test</scope>
```
[the remark this answers is not in the document]
Comment (jillingk) [reply pom.xml:201]: So outside of test, we would use scope>compile (default) for the other dependencies right?

[the remark this answers is not in the document]
Comment (wboereboom) [reply pom.xml:201]: Yes!

**Review by wboereboom — DISMISSED** [review]

**Review by wboereboom — APPROVED** [review]

**Review by michaelpaul — APPROVED** [review]

**Pull request merged**